File: student-teacher/src/api/courseAdjust.ts
import { http, normalizeList } from '@/utils/request'

/** 调课申请（course_adjust；status：待审核/已通过/已驳回/已撤销） */
export interface CourseAdjust {
  id?: number
  teacherId?: number
  courseId: number
  classId?: number
  /** 原上课日期 YYYY-MM-DD */
  originDate: string
  /** 原上课时段 */
  originSlot: string
  /** 调整后上课日期 */
  targetDate: string
  /** 调整后上课时段 */
  targetSlot: string
  /** 调整后教室ID（可空表示不变） */
  classroomId?: number
  reason?: string
  status?: string
  approveRemark?: string
  createTime?: string
  /** 展示用扩展字段（后端 Service 关联填充） */
  teacherName?: string
  courseName?: string
  className?: string
  roomName?: string
}

/**
 * 教师：提交调课申请
 * 后端 POST /api/course-adjusts（teacherId 由服务端按登录人写入）
 */
export function submitCourseAdjust(payload: CourseAdjust) {
  return http.post<null>('/course-adjusts', payload)
}

/**
 * 教师：我的调课（服务端按当前登录教师过滤）
 * 后端 GET /api/course-adjusts/my
 */
export async function listMyCourseAdjusts(): Promise<CourseAdjust[]> {
  const data = await http.get<unknown>('/course-adjusts/my')
  return normalizeList<CourseAdjust>(data)
}

/**
 * 教师：撤销自己的调课申请（仅待审核可撤销）
 * 后端 PUT /api/course-adjusts/{id}/cancel
 */
export function cancelCourseAdjust(id: number) {
  return http.put<null>(`/course-adjusts/${id}/cancel`)
}

/**
 * 管理端：审批列表（status 为空查全部）
 * 后端 GET /api/course-adjusts?status=
 */
export async function listCourseAdjusts(status?: string): Promise<CourseAdjust[]> {
  const data = await http.get<unknown>('/course-adjusts', status ? { status } : undefined)
  return normalizeList<CourseAdjust>(data)
}

File: edumanage/springboot/src/main/java/com/keshe/edumanage/entity/base/CourseAdjust.java
package com.keshe.edumanage.entity.base;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.keshe.edumanage.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

/**
 * 调课申请
 * <p>教师提交「原上课时间 → 调整后时间」的调课申请，由教务处审批。</p>
 */
@Data
@TableName("course_adjust")
@EqualsAndHashCode(callSuper = true)
public class CourseAdjust extends BaseEntity {
    /**
     * 申请教师ID
     */
    private Long teacherId;

    /**
     * 课程ID
     */
    private Long courseId;

    /**
     * 班级ID
     */
    private Long classId;

    /**
     * 原上课日期
     */
    private LocalDate originDate;

    /**
     * 原上课时段
     */
    private String originSlot;

    /**
     * 调整后上课日期
     */
    private LocalDate targetDate;

    /**
     * 调整后上课时段
     */
    private String targetSlot;

    /**
     * 调整后教室ID（可为空，表示教室不变）
     */
    private Long classroomId;

    /**
     * 调课原因
     */
    private String reason;

    /**
     * 状态：待审核/已通过/已驳回/已撤销
     */
    private String status;

    /**
     * 审批意见
     */
    private String approveRemark;

    /**
     * 以下为展示用字段（不属于 course_adjust 表），由 Service 关联填充。
     */
    @TableField(exist = false)
    private String teacherName;

    @TableField(exist = false)
    private String courseName;

    @TableField(exist = false)
    private String className;

    @TableField(exist = false)
    private String roomName;
}
